test/cmd/create.sh

```diff
@@ -153,3 +153,63 @@ run_kubectl_create_kustomization_directory_tests() {
   set +o nounset
   set +o errexit
 }
+
+# Runs tests related to kubectl create --validate
+run_kubectl_create_validate_tests() {
+  set -o nounset
+  set -o errexit
+
+  ## test --validate=true
+  create_and_use_new_namespace
+  kube::log::status "Testing kubectl create --validate=true"
+  # create and verify
+  output_message=$(! kubectl create -f hack/testdata/invalid-deployment-unknown-and-duplicate-fields.yaml --validate=true 2>&1)
+  kube::test::if_has_string "${output_message}" 'error validating data'
+
+  ## test  --validate=false
+  kube::log::status "Testing kubectl create --validate=false"
+  # create and verify
+  output_message=$(kubectl create -f hack/testdata/invalid-deployment-unknown-and-duplicate-fields.yaml --validate=false)
+  kube::test::if_has_string "${output_message}" "deployment.apps/invalid-nginx-deployment created"
+  # cleanup
+  kubectl delete deployment invalid-nginx-deployment
+
+  ## test --validate=strict
+  create_and_use_new_namespace
+  kube::log::status "Testing kubectl create --validate=strict"
+  # create and verify
+  output_message=$(! kubectl create -f hack/testdata/invalid-deployment-unknown-and-duplicate-fields.yaml --validate=strict 2>&1)
+  kube::test::if_has_string "${output_message}" 'error validating data'
+
+  ## test --validate=warn
+  create_and_use_new_namespace
+  kube::log::status "Testing kubectl create --validate=warn"
+  # create and verify
+  output_message=$(! kubectl create -f hack/testdata/invalid-deployment-unknown-and-duplicate-fields.yaml --validate=warn 2>&1)
+  kube::test::if_has_string "${output_message}" 'cannot perform warn validation if fieldValidation parameter is unsupported'
+
+  ## test  --validate=ignore
+  kube::log::status "Testing kubectl create --validate=ignore"
+  # create and verify
+  output_message=$(kubectl create -f hack/testdata/invalid-deployment-unknown-and-duplicate-fields.yaml --validate=ignore)
+  kube::test::if_has_string "${output_message}" "deployment.apps/invalid-nginx-deployment created"
+  # cleanup
+  kubectl delete deployment invalid-nginx-deployment
+
+  ## test default is strict validation
+  create_and_use_new_namespace
+  kube::log::status "Testing kubectl create"
+  # create and verify
+  output_message=$(! kubectl create -f hack/testdata/invalid-deployment-unknown-and-duplicate-fields.yaml 2>&1)
+  kube::test::if_has_string "${output_message}" 'error validating data'
+
+  ## test invalid validate value
+  create_and_use_new_namespace
+  kube::log::status "Testing kubectl create --validate=foo"
+  # create and verify
+  output_message=$(! kubectl create -f hack/testdata/invalid-deployment-unknown-and-duplicate-fields.yaml --validate=foo 2>&1)
+  kube::test::if_has_string "${output_message}" 'Invalid validate value'
+
+  set +o nounset
```

Comment:
There should be a clean up for `create_and_use_new_namespace`. Deployment seems ok but for namespace which is created already?

Reply:
oops, all these extra `create_and_use_new_namespace` calls are erroneous (copied from when I was duplicating test cases). I've removed them, so that there's only one `create_and_use_new_namespace` at the beginning of the test. I don't think we need to perform an explicit clean up though, because most tests seem to call it at the beginning of the test without a cleanup for it.